samples/cpp/tutorial_code/ImgTrans/HoughCircle_Demo.cpp

```diff
@@ -91,7 +91,7 @@ int main(int argc, char** argv)
     // and refresh the content of the output image
     // until the user presses q or Q
     int key = 0;
-    while(key != 'q' && key != 'Q')
+    while((char) key != 'q' && (char) key != 'Q')
```

Comment:
You should not truncate key value. This range exceed values of standard char "[-128;127]".
```
(char)0x131 == '1' /*0x31*/
```
is still "true", but it is wrong.
You should cast both to bigger value ("int" in this case) or try to change compiler options and disable noise warnings.